packages/dds/tree/src/test/simple-tree/treeApi.spec.ts

```diff
@@ -768,5 +768,147 @@ describe("treeNodeApi", () => {
 			assert.equal(nodeChanged, true, "'nodeChanged' should have fired");
 			assert.equal(treeChanged, true, "'treeChanged' should have fired");
 		});
+
+		it(`'nodeChanged' includes the names of changed properties (objectNode)`, () => {
+			const sb = new SchemaFactory("test");
+			class treeSchema extends sb.object("root", {
+				prop1: sb.optional(sb.number),
+				prop2: sb.optional(sb.number),
+				prop3: sb.optional(sb.number),
+			}) {}
+
+			const view = getView(new TreeViewConfiguration({ schema: treeSchema }));
+			view.initialize({ prop1: 1, prop2: 2 /* , prop3: 0 */ });
+			const root = view.root;
+
+			let listenerFired = false;
+			Tree.on(root, "nodeChanged", ({ changedProperties }) => {
+				// This is the main validation for the test
+				assert.deepEqual(changedProperties, new Set(["prop1", "prop2", "prop3"]));
+				listenerFired = true;
+			});
```

Comment:
this is fine, but I think a cleaner and more general pattern for this kind of test is something along the lines of:
```typescript
const log: Set<string> = [];
Tree.on(root, "nodeChanged", ({ changedProperties }) => log. push(changedProperties)
assert.deepEqual(log, [new Set(["prop1", "prop2", "prop3"])]);
```

This pattern is more concise, detects if the callback is called more than once, and provides more context when erroring (ex: if its called multiple times with different data, you get a single error showing that all at once).

Also you can reset the log with `log.length= 0` and then run another scenario in the same test without having to setup a new callback so you can test several simple cases concicesly.

Reply:
Love it, updating the tests.